wagtailio/core/models/pages.py

```diff
@@ -130,3 +138,165 @@ class ContentPage(Page, HeroMixin, SocialMediaMixin, CrossPageMixin):
         index.SearchField("intro"),
         index.SearchField("body"),
     ]
+
+
+class ShowcasePage(Page, SocialMediaMixin, CrossPageMixin):
+    template = "patterns/pages/showcase_page/showcase_page.html"
+    htmx_template = "patterns/pages/showcase_page/showcase_page_fragment.html"
+
+    parent_page_types = ["core.HomePage"]
+
+    introduction = models.TextField(
+        verbose_name="Introduction",
+        blank=True,
+    )
+    cta = StreamField(
+        [("cta", StandaloneCTABlock())],
+        blank=True,
+        max_num=1,
+        help_text="Allows for a maximum of 1 CTA blocks",
+        use_json_field=True,
+    )
+
+    content_panels = Page.content_panels + [
+        FieldPanel("introduction"),
+        FieldPanel("cta"),
+        InlinePanel("showcase_items", label="Showcase items"),
+    ]
+
+    promote_panels = (
+        Page.promote_panels + SocialMediaMixin.panels + CrossPageMixin.panels
+    )
+
+    search_fields = Page.search_fields + [
+        index.SearchField("description"),
+        index.SearchField("introduction"),
+    ]
+
+    def _is_htmx_request(self, request):
+        return request.headers.get("HX-Request") == "true"
+
+    def _filter_used_sectors(self):
+        return ["All"] + list(
+            self.showcase_items.values_list("sector__name", flat=True).distinct()
+        )
+
+    def _filtered_showcase_items(self, sector):
+        if sector in self._filter_used_sectors() and sector != "All":
+            return self.showcase_items.filter(sector__name=sector)
+        else:
+            return self.showcase_items.all()
+
+    def get_template(self, request, *args, **kwargs):
+        if self._is_htmx_request(request):
+            return self.htmx_template
+        else:
+            return self.template
+
+    def serve(self, request):
+        response = super().serve(request)
+        if self._is_htmx_request(request):
+            # We only return the fragment, rather than the whole page.
+            new_url = self.url + "?" + request.GET.urlencode()
+            response.headers["HX-Push"] = new_url
```

Comment:
You can also add https://htmx.org/headers/hx-push-url/ and avoid the inline script in the template fragment

Reply:
Sure thing, this is neat.